### PR TITLE
Add new assertion type "contains"

### PR DESCRIPTION
**Pull request by JakeChampion**
This adds the feature requested in #114.

### CODE REVIEW

**Comment by mirisuzanne** [comment]
Thanks, this looks really cool! I may leave a few review comments, if you're open to doing some final cleanup before we merge?

**Comment by JakeChampion** [comment]
I am happy to do some final cleanup :-)

File: test/scss/_assert.scss [hunk 1]
```diff
@@ -24,8 +24,8 @@
 
   @include test('False is not truthy') {
     @include assert-equal(
-      _true-is-truthy(false),
-      false);
```
Comment (mirisuzanne) [inline test/scss/_assert.scss]: Can you run `gulp sasslint` to check for unnecessary white-space and other minor code-formatting cleanup? You can also use `npm run commit` to lint, test, and compile docs in one command.

File: test/scss/_assert.scss [hunk 2]
```diff
@@ -217,4 +215,139 @@
...
+            }
+          }
+        }
+      }
+    }
+
+    @include test('Can be used for all selector types') {
+      @include assert {
+        @include output {
+          .selector {
```
Comment (mirisuzanne) [inline test/scss/_assert.scss]: I imagine this also works with attribute selectors, and pseudo-selectors? May be worth adding tests for those? What about *, >, ~, etc?

File: sass/true/_assert.scss [hunk 3]
```diff
@@ -317,6 +317,60 @@
...
+///     @include assert {
+///       @include output {
+///         height: 100%;
+///         width: 14em + 2;
+///       }
+///       @include contains {
+///         width: 16em;
+///       }
+///     }
+///   }
```
Comment (jgerigmeyer) [inline sass/true/_assert.scss]: This block triggers a warning during `npm run sassdoc` compilation:

```
[output-context] contains is not a valid context for output tests: assert expect output
```

Changes are required in `_context.scss` (`_true-output-context` and `_true-validate-output-context` functions) to allow `contains` as a valid context.

File: lib/main.js [hunk 4]
```diff
@@ -305,10 +312,135 @@ var parse = module.exports.parse = function (rawCss, contextLines) {
...
+
+
   return parseCss();
 };
 
+var flatten = function flatten(list) {
+  return list.reduce(function (a, b) {
+      return a.concat(Array.isArray(b) ? flatten(b) : b);
+  }, []);
+};
```
[the remark this answers is not in the document]
Comment (JakeChampion) [reply lib/main.js]: Thanks for the review :-) It looks like it is not required after the code was refactored.

File: test/scss/_assert.scss (hunk 2 above)
Comment (JakeChampion) [reply test/scss/_assert.scss]: It works for these selectors :-)

File: test/scss/_assert.scss [hunk 5]
```diff
@@ -217,4 +215,191 @@
...
+          }
+
+          * + * {
+            display: none;
+          }
+        }
+      }
+    }
+
+    @include test('Can assert multiple properties within a selector') {
```
Comment (JakeChampion) [inline test/scss/_assert.scss]: This is the failing test right now, will look into how to solve this issue.

File: sass/true/_context.scss [hunk 6]
```diff
@@ -87,8 +87,8 @@ $_true-output-context: ();
 /// Validate the new context, and return an updated context value
 /// @access private
 /// @group private-context
-/// @param {'assert' | 'output' | 'expect' | null} $new -
-///   Add a new `assert`, `output`, or `expect` layer
+/// @param {'assert' | 'output' | 'expect' | 'contains' | null} $new -
+///   Add a new `assert`, `output`, `expect`, or `contains` layer
```
Comment (jgerigmeyer) [inline sass/true/_context.scss]: I may be mistaken, but I think we also want this type of change on L79, L106, L120, L121, L129, and L132.

**Comment by JakeChampion** [comment]
New tests added, documentation updated and a bug was spotted and fixed :-)

File: sass/true/_context.scss (hunk 6 above)
Comment (jgerigmeyer) [reply sass/true/_context.scss]: Again, correct me if I'm wrong, but I think this change (adding `contains`) should also happen on L64, L79, L100, L120, and L121.

**Comment by JakeChampion** [comment]
@jgerigmeyer Thanks for pointing those issues out, they should now be fixed

**Comment by JakeChampion** [comment]
@mirisuzanne, @jgerigmeyer is there anything else that needs to be done by myself for this to be merged and released? :-)

**Comment by mirisuzanne** [comment]
@jackmcpickle Nope! I think it's back in my court - and I can probably get to it tomorrow. Thanks!

**Pull request merged**